src/bin/pgcopydb/ld_replay.c

```diff
@@ -249,3 +250,136 @@ stream_replay_line(void *ctx, const char *line, bool *stop)
 
 	return true;
 }
+
+
+/* streamReplayLineIntoBuffer is a callback function for the ReadFromStreamContext
+ * and read_from_stream infrastructure. It's called on each line read from a
+ * stream such as a unix pipe and buffers the line into a file until we encounter
+ * a COMMIT. This ensures that we can replay the whole transaction in one go
+ * and avoids deadlock when the BEGIN messages arrives before the existence of
+ * transation metadata file which is created by the transform process.
+ */
+static bool
+streamReplayLineIntoBuffer(void *ctx, const char *line, bool *stop)
+{
+	ReplayStreamCtx *replayCtx = (ReplayStreamCtx *) ctx;
+	StreamApplyContext *context = &(replayCtx->applyContext);
+
+	if (strstr(line, OUTPUT_BEGIN) == line)
+	{
```

Comment:
Could we maybe call `parseSQLAction` first and then use the metadata action field to sort out what's happening here, rather than comparing strings again?

Reply:
I actually avoided because it allocates memory for jsonBuffer for EXECUTE message which we never use. I will change it to use parseSQLAction.

Reply:
See next series of comments. Resolving this one.